### DISCUSSION TITLE
CP2K ELPA diagonalization wrapper upgraded to require ELPA 2018.05.001+

### DISCUSSION THREAD

**Discussion by dev-zero** [cp2k/cp2k#2591] [Team Posts]
Dear all,

I've just merged my [PR #477](https://github.com/cp2k/cp2k/pull/477) which adapts our wrapper for ELPA's diagonalization routines to the "new" (introduced in 2017) ELPA API. The minimal required version of ELPA is now 2018.05.001.
This will be checked at runtime, meaning that while you can link CP2K against 2017 <= ELPA < 2018.05.001, it will fail at runtime.
Furthermore it is not necessary anymore to specify the version of ELPA during the compilation: instead of `-D__ELPA=20xxyyzz` it is sufficient to write `-D__ELPA`.

Best regards,
Tiziano

**Comment by rathann** [comment]
That is really cool, although I would much prefer that it failed at build time. Also, ELPA 2019.05.001 is out. Is it supported already?

**Comment by mkrack** [comment]
The parallel PSI/Merlin regression tester are built all with ELPA 2019.05.001 and they are OK.
